google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClient.java

```diff
@@ -311,6 +403,33 @@ public abstract static class Builder {
     /** Whether discovery required parameter checks should be suppressed. */
     boolean suppressRequiredParameterChecks;
 
+    /** User configured Universe Domain. Defaults to `googleapis.com`. */
+    String universeDomain;
+
+    /**
+     * Regex pattern to check if the URL passed in matches the default endpoint confgured from a
+     * discovery doc. Follows the format of `https://{serviceName}(.mtls).googleapis.com/`
+     */
+    Pattern defaultEndpointRegex =
+        Pattern.compile("https://[a-zA-Z]*(\\.mtls)?\\.googleapis.com/?");
```

Comment:
I did a few quick test but I don't think this regex works, what worked for me is 
```https:\/\/[a-zA-Z]*(\.mtls)?\.googleapis.com\/?```
Can you please make sure the regex is correct and add a few tests just for the regex?

Reply:
Oh huh, I did add unit tests for this regex and it did pass: https://github.com/googleapis/google-api-java-client/blob/1cfd93ae34f44932b135a9dafa1fa6d8698448ae/google-api-client/src/test/java/com/google/api/client/googleapis/services/AbstractGoogleClientTest.java#L328-L357.  Do you have the values you tested with? I may have tested incorrectly.